### PR TITLE
Fix  #7619: Wallet V2 onboarding flow

### LINKED ISSUE #7619: Wallet: V2 Create a Wallet
```[tasklist]
### Tasks
- [x] Wallet: V2 Welcome
- [x] Wallet: V2 Create a new Password
- [x] Wallet: V2 Back up Recovery Phrase
- [x] Wallet: V2 Verify Recovery Phrase
- [x] Wallet: V2 Biometrics
- [x] Wallet: V2 Confirmation
- [x] Wallet: V2 Restore Wallet
```

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
Figma link: https://www.figma.com/file/oXIiv45pIhF2RGlpEjOfUg/Wallet-onboarding?type=design&node-id=3271%3A119412&mode=design&t=PHVbewOicYd43G1R-1

This pull request fixes #7619

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Test Plan:

Please check all the flow according to the figma link
Please check all the error states
Please check if back up recovery phrases after creating a wallet still work inside wallet

## Screenshots:

1. Create a new wallet without backup in back up recovery phrase screen

https://github.com/brave/brave-ios/assets/1187676/eb9e88e0-af3a-4286-987b-6758b7cdd8ef

2. Back up recovery phrase after wallet is create

https://github.com/brave/brave-ios/assets/1187676/7f4bb8e0-8b97-4f9b-9e39-752db7bcb08a

3. Create a new wallet without backup in verify recovery phrase screen

https://github.com/brave/brave-ios/assets/1187676/b54bbc48-cf13-48e2-a842-b60fdd93f304

4. Create a new wallet and back up recovery phrase but biometric login is not available 

https://github.com/brave/brave-ios/assets/1187676/2619998e-067d-4573-a146-c0b644a6c106

5. Create a new wallet and back up recovery phrase and biometric login is available 

https://github.com/brave/brave-ios/assets/1187676/3f126de7-2d75-4a39-b04f-6d20ff0e2f8c

7. Restore a wallet via recovery phrase

https://github.com/brave/brave-ios/assets/1187676/6c726a84-e322-4134-ad8c-e7d5eab7a4c9

8. Restore a wallet via recovery phrase error state 

https://github.com/brave/brave-ios/assets/1187676/c0edc278-2e46-4832-a287-4723f0e92fd1

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Sources/BraveWallet/WalletStrings.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/WalletStrings.swift]: Click? I think we'd want `Tap` for mobile 🙂

File: Sources/BraveWallet/Assets.xcassets/Brave Wallet/Onboarding/wallet-add.imageset/add@2x.png
Comment (StephenHeaps) [inline Sources/BraveWallet/Assets.xcassets/Brave Wallet/Onboarding/wallet-add.imageset/add@2x.png:1]: Maybe we can use [`leo.plus.add` SF Symbol](https://github.com/brave/leo-sf-symbols/blob/main/symbols/leo.plus.add.svg) with a purple tint instead of this png? cc @minkyeongshin

File: Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift]: Do these need localized?

File: Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift]:
```suggestion
    _password = State(initialValue: restorePackage?.newPassword ?? "")
    _repeatedPassword = State(initialValue: restorePackage?.newPassword ?? "")
```

File: Sources/BraveWallet/Crypto/Onboarding/RecoveryPhrasePager.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/RecoveryPhrasePager.swift]: nit: indent is off

File: Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift]:
```suggestion
          recoveryWords = .init(repeating: "", count: isLegacyWallet ? 12 : 24)
```

Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift]: Hmm actually, `isLegacyWallet` computed property is 24 characters, one of these seems to be backwards?
https://github.com/brave/brave-ios/blob/78bdbe4b84453e535b4053d36190993bbdd384b9/Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift#L58-L60

File: Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift]: We probably want to safely access the values in this array? 🤔

File: Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift]: Maybe this could be a computed property on the `RestorePackage`? Or centralized somewhere... we seem to have this 24 / 12 in a few spots, but extension on `Array` where element is `String` seems to generic.

File: Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift]: nit: maybe we could make an extension for this that just accepts a `backButtonTitle: String`?

File: Sources/BraveWallet/Assets.xcassets/Brave Wallet/Onboarding/wallet-add.imageset/add@2x.png
Comment (nuo-xu) [reply Sources/BraveWallet/Assets.xcassets/Brave Wallet/Onboarding/wallet-add.imageset/add@2x.png:1]: the reason i added a new image instead of using `leo.plus.add` is because the import wallet icon in the same screen has no SF symbol in leo system. i tried to use one SF symbol and one image but i'm having a hard time to align the both sizes. if @minkyeongshin can add the other import wallet icon to the leo SF symbol, and we can use both icons as the leo SF Symbol is the best solution imo.

File: Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift]: this is actually correct since `isLegacyWallet` meaning the user currently is inputting 24 recovery phrases and this button is to change to the not legacy (regular) wallet which only has 12 recovery phrases. i will add comment to eliminate the confusion.

File: Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift]: added a computed boolean inside `RestorePackage`. also centralized the 12/24 as an extension of `Int` in [16b04a0](https://github.com/brave/brave-ios/pull/7746/commits/16b04a0eb46deb563357557b0a34e41a60d1321b)

File: Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift]: nit: 
```suggestion
  @State private var isNewWalletCreated: Bool = false
```

File: Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift:1]: We'll need to obfuscate the word length when we blur the recovery phrase similar to @stoletheminerals previous PR: https://github.com/brave/brave-ios/pull/7489

File: Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift]: We might want to reduce the horizontal padding on this view (this might be incorrect line for the padding, but the padding left/right of the recovery phrase box), I've had a word wrap a single character at the default font size a couple times in testing this PR.

File: Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift:55]: I think this field needs a placeholder, it doesn't look enough like a field without it.
Simulator Screenshot - iPhone 14 Pro - 2023-07-20 at 10 29 11

File: Sources/BraveWallet/WalletStrings.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/WalletStrings.swift]: This also changes the title when the full native wallet is open, is that intended? Desktop just has `Wallet` page title so maybe fine, just checking 🙂

File: Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift]: whitespace warning here

File: Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift
Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift]: First comment still an issue (helps with line length) but second comment resolved nicely, thanks 👍

File: Sources/BraveWallet/WalletStrings.swift
Comment (nuo-xu) [reply Sources/BraveWallet/WalletStrings.swift]: There was no screen in onboarding flow has a nav title. This was accidentally done with an old design. 
I'm seeing design changed `Brave Wallet` to `Wallet` in the new portfolio screen. I will revert the change here and leave it to Portfolio V2 PR in the future.

File: Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift
Comment (StephenHeaps) [reply Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift:1]: Actually I think this might be ok, I was mistaken here. @stoletheminerals fix was when the screen is being recorded, or app is backgrounded (app switcher). That logic still applies here:
Simulator Screenshot - iPhone 14 Pro - 2023-07-21 at 10 51 24

File: Sources/BraveWallet/Assets.xcassets/Brave Wallet/Onboarding/wallet-add.imageset/add@2x.png
Comment (nuo-xu) [reply Sources/BraveWallet/Assets.xcassets/Brave Wallet/Onboarding/wallet-add.imageset/add@2x.png:1]: The newest Leo SF Symbol has the import symbol. I have upgraded the dependency and used the symbols for both places in [1d4972d](https://github.com/brave/brave-ios/pull/7746/commits/1d4972df480781527014d1a81b091713712cdd90)

File: Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift]: updated in [441e43e](https://github.com/brave/brave-ios/pull/7746/commits/441e43ec278e55dc9d526b3f5c2d00ed48ca50c0)

File: Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift]: updated in [441e43e](https://github.com/brave/brave-ios/pull/7746/commits/441e43ec278e55dc9d526b3f5c2d00ed48ca50c0)

File: Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/BackupRecoveryPhraseView.swift]: updated in [1d4972d](https://github.com/brave/brave-ios/pull/7746/commits/1d4972df480781527014d1a81b091713712cdd90)

File: Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift:55]: as discussed we will auto-focused on this field once user comes to this screen. Same as the first password field once user lands in create a new password screen. updated in [441e43e](https://github.com/brave/brave-ios/pull/7746/commits/441e43ec278e55dc9d526b3f5c2d00ed48ca50c0)

File: Sources/BraveWallet/WalletStrings.swift
Comment (nuo-xu) [reply Sources/BraveWallet/WalletStrings.swift]: reverted in [1d4972d](https://github.com/brave/brave-ios/pull/7746/commits/1d4972df480781527014d1a81b091713712cdd90)

File: Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift]: updated in [441e43e](https://github.com/brave/brave-ios/pull/7746/commits/441e43ec278e55dc9d526b3f5c2d00ed48ca50c0)

File: Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift:87]: We should reset `passwordToSaveInBiometric` here to nil after saving successfully (and unsuccessfully?). Ideally we avoid storing in `KeyringStore` entirely if we can, but not sure it's simple to pass from view to view in this flow 🤔.

**Review by StephenHeaps — COMMENTED** [review]
Resolved by hiding navigation bar back button during onboarding (not hidden when backing up later 👍)

~Sharing this one early while I continue review. The wallet is created after entering & confirming password, but if you tap back, then tap `Continue` again it creates the wallet again. However when you complete onboarding, you'll have a Solana account for each time the wallet is created / `Continue` is tapped on `Create Password` screen:~


https://github.com/brave/brave-ios/assets/5314553/11a01282-a781-43f8-aed7-b16f12e36261

File: Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift:87]: it's now set to nil after its saved or failed to saved to keychain. [0de541c](https://github.com/brave/brave-ios/pull/7746/commits/0de541cedd3b21b3d9c244c267b31204cd4b6b7e)

File: Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift]:
```suggestion
  private var isContinueDisabled: Bool {
    recoveryWords.allSatisfy({ !$0.isEmpty })
  }
```

nit: I believe you can just use `allSatisfy` here because `recoveryWords` array will either be updated to the correct length when switching from 12/24 words 🙂

File: Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/CreateWalletView.swift]: nit: `isFieldFocused`

File: Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Onboarding/VerifyRecoveryPhraseView.swift]: nit: `isFieldFocused`

**Review by StephenHeaps — COMMENTED** [review]
Resolved by hiding skip button when backing up seed phrase after onboarding via banner / Accounts `Backup` button 👍.

~Opening Backup via the Portfolio banner / Accounts `Backup` button the `Skip` -> `I'll backup later` button isn't working as intended:~

https://github.com/brave/brave-ios/assets/5314553/c01723a7-1473-4234-815b-2746a90540a6

Everything else resolved, just a couple nits 🙂.

File: Sources/BraveUI/SwiftUI/ScrollViewIndicatorFlash.swift
Comment (StephenHeaps) [inline Sources/BraveUI/SwiftUI/ScrollViewIndicatorFlash.swift]: Just realized this doesn't seem to be used anywhere, maybe after change to show all 24 words w/o scrolling?

**Review by StephenHeaps — APPROVED** [review]
Really nice work on this! New onboarding looks great 🐎.

File: Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift
Comment (kdenhartog) [inline Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift:85]: just double checking for my sanity - is the keystore in the OS keychain sandboxed per app?

Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Onboarding/BiometricView.swift:85]: an application can access only its own keychain items as it depends on app's bundle id and the id is unique.

**Review by StephenHeaps — APPROVED** [review]
I think the toggles on the Legal page should be on the right or let users tap on the text to activate the toggle (desktop allows this), but I'll leave that for design 👍

**Review by kdenhartog — APPROVED** [review]
Sorry, just realized I didn't actually add the approval on the PR and only added the label on the sec review issue. The legal text has been added so all [security blockers](https://github.com/brave/reviews/issues/1348#issuecomment-1659487902) have been addressed. LGTM

**Pull request merged**
